inference-engine/src/mkldnn_plugin/nodes/mkldnn_space_to_depth_node.cpp

```diff
@@ -139,108 +127,134 @@ void MKLDNNSpaceToDepthNode::initSupportedPrimitiveDescriptors() {
     }
     supportedTypes.push_back(LayoutType::ncsp);
     auto creators = BlockedDescCreator::getCommonCreators();
-    auto range = BlockedDescCreator::makeFilteredRange(creators, nDims, supportedTypes);
+    auto range = BlockedDescCreator::makeFilteredRange(creators, inputDataShape.getRank(), supportedTypes);
 
     for (auto itr = range.first; itr != range.second; ++itr) {
-        config.inConfs[0].desc = itr->second->createSharedDesc(precision, getInputShapeAtPort(0));
-        config.outConfs[0].desc = itr->second->createSharedDesc(precision, getOutputShapeAtPort(0));
+        config.inConfs[0].desc = itr->second->createSharedDesc(precision, inputDataShape);
+        config.outConfs[0].desc = itr->second->createSharedDesc(precision, outputDataShape);
         supportedPrimitiveDescriptors.emplace_back(config, impl_type);
     }
 }
 
 void MKLDNNSpaceToDepthNode::createPrimitive() {
-    auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
-    auto &srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
+    auto& dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    auto& srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
     if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
         THROW_ERROR << "has not allocated destination memory";
     if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
         THROW_ERROR << "has not allocated input memory";
     if (getSelectedPrimitiveDescriptor() == nullptr)
         THROW_ERROR << "has unidentified preferable primitive descriptor";
 
-    VectorDims srcDims = srcMemPtr->getStaticDims();
-    VectorDims dstDims = dstMemPtr->getStaticDims();
+    const auto& memoryDesc = srcMemPtr->getDesc();
+    attrs.dataSize = memoryDesc.getPrecision().size();
+    attrs.nSpatialDims = memoryDesc.getShape().getRank() - 2;
```

Comment:
better to initialize `nSpatialDims` in ctor becuase you initialized `blockStep` which depends on `nSpatialDims`
 `attrs.blockStep = static_cast<size_t>(std::pow(attrs.blockSize, nSpatialDims));`